File: Wordle/GuessResult.cs
using System;
using System.Collections.Generic;

namespace Wordle
{

	public class GuessResult
	{
        public List<LetterGuess> Guess { get; set; }

        public GuessResult(string guess)
		{
			Guess = new List<LetterGuess>(guess.Length);

			foreach(var letter in guess)
            {
				Guess.Add(new LetterGuess(letter));
            }
		}

        public override string ToString()
        {
            string result = "";

            foreach(var item in Guess)
            {
                result += item.Letter;
            }

            result += "\n";

            foreach (var item in Guess)
            {
                switch(item.LetterResult)
                {
                    case LetterResult.Correct:
                        result += "C";
                        break;

                    case LetterResult.Misplaced:
                        result += "M";
                        break;

                    case LetterResult.Incorrect:
                        result += "I";
                        break;
                }
            }

            result += "\n";

            return result;
        }
    }
}

File: pywordle/guess_result.py
from letter_guess import LetterResult
from letter_guess import LetterGuess


class GuessResult:
    def __init__(self, guess: str):
        self.guess = []

        for letter in guess:
            self.guess.append(LetterGuess(letter))

    def __str__(self):
        result = ""

        for letter in self.guess:
            result += letter.letter

        result += "\n"

        for letter in self.guess:
            if letter.letter_result == LetterResult.correct:
                result += "C"
            elif letter.letter_result == LetterResult.misplaced:
                result += "M"
            else:
                result += "I"

        result += "\n"

        return result
